Moved timeseries peek to plot()

jeffreypaul15 opened the pull request:
<!--
We know that working on code and submitting pull requests takes effort, and we appreciate your time.
Thank you.

Please be aware that everyone has to follow our code of conduct:
https://sunpy.org/coc

Furthermore, you might need to check with your work place if you are allowed to contribute code:
https://docs.sunpy.org/en/latest/dev_guide/contents/newcomers.html

Also these comments are hidden when you submit this github pull request.

We have several automatic features that run when a pull request is open.
They can appear daunting but do not worry about them!
Here is a brief explanation of them:
https://docs.sunpy.org/en/latest/dev_guide/contents/pr_review_procedure.html#continuous-integration
-->

### Description

<!--
Provide a general description of what your pull request does.

If the pull request closes any open issues you can add this.
If you replace <Issue Number> with a number GitHub will automatically link it.
If it doesn't, please remove the following line.
-->

Fixes #5066 
Drawing parallels from GenericMap, I suppose this is along the lines of what we want?
This does override the base `plot()` method defined for TimeSeries though.

pep8speaks commented:
Hello @jeffreypaul15! Thanks for opening this PR. 

* In the file [`sunpy/timeseries/sources/goes.py`](https://github.com/sunpy/sunpy/blob/80b21094f06c7c0ddde679d2b3bd5e07f820a628/sunpy/timeseries/sources/goes.py):

> [Line 215:101](https://github.com/sunpy/sunpy/blob/80b21094f06c7c0ddde679d2b3bd5e07f820a628/sunpy/timeseries/sources/goes.py#L215): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 145:101](https://github.com/sunpy/sunpy/blob/80b21094f06c7c0ddde679d2b3bd5e07f820a628/sunpy/timeseries/sources/goes.py#L145): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (111 > 100 characters)
> [Line 110:101](https://github.com/sunpy/sunpy/blob/80b21094f06c7c0ddde679d2b3bd5e07f820a628/sunpy/timeseries/sources/goes.py#L110): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 73:19](https://github.com/sunpy/sunpy/blob/80b21094f06c7c0ddde679d2b3bd5e07f820a628/sunpy/timeseries/sources/goes.py#L73): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 71:22](https://github.com/sunpy/sunpy/blob/80b21094f06c7c0ddde679d2b3bd5e07f820a628/sunpy/timeseries/sources/goes.py#L71): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 44:101](https://github.com/sunpy/sunpy/blob/80b21094f06c7c0ddde679d2b3bd5e07f820a628/sunpy/timeseries/sources/goes.py#L44): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)

Do see the [Hitchhiker's guide to code style](https://goo.gl/hqbW4r)

jeffreypaul15 commented:
@ayshih Is this something along the lines of what we want?

nabobalis commented:
This looks like something we want, tho It will need to be done for each time series source since they all have their own peek method.

jeffreypaul15 commented:
> This looks like something we want, tho It will need to be done for each time series source since they all have their own peek method.

Yeah, I just made the change for this one source to confirm if this is what we want. Shall I go about with the rest?

nabobalis commented:
Sure

jeffreypaul15 commented:
@nabobalis I've added all the TimeSeries sources, but like I mentioned, this does overwrite the `plot()` method defined in the base TimeSeries class, how would we want to handle that?

nabobalis commented:
> @nabobalis I've added all the TimeSeries sources, but like I mentioned, this does overwrite the `plot()` method defined in the base TimeSeries class, how would we want to handle that?

You mean it overloads it?

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -18,7 +18,7 @@
 from sunpy.visualization import peek_show
 
-__all__ = ['EVESpWxTimeSeries', 'ESPTimeSeries']
+__all__ = ["EVESpWxTimeSeries", "ESPTimeSeries"]
```
Did you run a linter?

jeffreypaul15 replied:
No I ran codestyle on it, isn't this what we should be using instead?

nabobalis replied:
What do you mean? The code style shouldn't make any of these changes.

jeffreypaul15 replied:
Oh yeah, my apologies. These were changed by `autopep8`.

jeffreypaul15 commented:
> You mean it overloads it?

No, I am overriding the plot method right

nabobalis commented:
> > You mean it overloads it?
> 
> No, I am overriding the plot method right

I don't see the issue. What would be the problem.

jeffreypaul15 commented:
> > > You mean it overloads it?
> > 
> > 
> > No, I am overriding the plot method right
> 
> I don't see the issue. What would be the problem.

We lose that functionality, basically the ability to call the base plot method. Is that fine?

nabobalis commented:
> > > > You mean it overloads it?
> > > 
> > > 
> > > No, I am overriding the plot method right
> > 
> > 
> > I don't see the issue. What would be the problem.
> 
> We lose that functionality, basically the ability to call the base plot method. Is that fine?

Don't see why not.

jeffreypaul15 commented:
What changelog would this PR require?

nabobalis commented:
> What changelog would this PR require?

Not sure. It feels kind of breaking to make plot do all this formatting but it doesn't actually break anything.

jeffreypaul15 commented:
> Not sure. It feels kind of breaking to make plot do all this formatting but it doesn't actually break anything.

Shall I go with feature then?

nabobalis commented:
Sure.

jeffreypaul15 commented:
> Sure.

Need to word the changelog better, please make the required changes.

jeffreypaul15 commented:
@nabobalis I don't suppose extra tests are needed right? The test for `peek()` covers `plot()` so would we need separate tests for `plot()`?

nabobalis commented:
> @nabobalis I don't suppose extra tests are needed right? The test for `peek()` covers `plot()` so would we need separate tests for `plot()`?

No it should be ok.

jeffreypaul15 commented:
@nabobalis Anything else to be done in this PR?

nabobalis commented:
It will need a review.

nabobalis commented on changelog/5200.feature.rst:
```diff
@@ -0,0 +1 @@
+Added `plot()` method to all `sunpy.timeseries.sources` and `peek()` now calls plot instead.
```
```suggestion
Added ``plot`` method to all `~sunpy.timeseries.Timeseries` sources and the ``peek`` method calls this instead.
This ensures that ``plot`` and ``peek`` will output figures that look the same.
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -69,23 +85,12 @@ def peek(self, title='EVE/ESP Level1', **kwargs):
         """
 
         self._validate_data_for_plotting()
```
Does this not belong in plot?

jeffreypaul15 replied:
Should this be moved there? It'd make sense I suppose.

nabobalis replied:
Well why would we only want to validate the data for peek but not plot?

jeffreypaul15 replied:
Yeah that makes sense, I'll make that change.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -69,23 +85,12 @@ def peek(self, title='EVE/ESP Level1', **kwargs):
-
-        names = ('Flux \n 0.1-7nm', 'Flux \n 18nm', 'Flux \n 26nm', 'Flux \n 30nm', 'Flux \n 36nm')
-
         axes = self.to_dataframe().plot(subplots=True, sharex=True, **kwargs)
```
Same here?

jeffreypaul15 replied:
Similar to how GenericMap is written, I've set `axes` in peek and then passed it to plot. Do we not want this?

nabobalis replied:
Does plot not need an axes?

jeffreypaul15 replied:
If plot is called directly it still gets the axes from `plt.gca()` right

nabobalis replied:
Then do we need this line?

jeffreypaul15 replied:
Sorry, I misunderstood this part. There is redundant code here, the reason I did this was in all sources `axes` was set in `peek` and then passed to `plot`. We don't use `plt.gca()` here but for the sake of uniformity I left it as it is, if plot is called directly, the same code is repeated there as well. I'll make the change if this is redundant.

nabobalis replied:
I am not sure, it feels redundant. You might have to have a play around to decide.

jeffreypaul15 replied:
This is the only source that behaves this way, the other sources axes aren't defined in this manner so I guess I'll make an exception for this?

nabobalis replied:
I would like to know why we make this exception for this source.

jeffreypaul15 replied:
This source fetches the `axes` from a pandas DataFrame object which returns a list of 2 axes.

nabobalis replied:
Fair enough.

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,6 +59,24 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
     # Class attribute used to specify the source class of the TimeSeries.
     _source = 'gbmsummary'
 
+    def plot(self, axes=None, **kwargs):
```
Docstring missing.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,8 +57,24 @@ class ESPTimeSeries(GenericTimeSeries):
 
     _source = 'esp'
 
+    def plot(self, axes=None, title="EVE/ESP Level1", **kwargs):
```
Is it not possible to call the generic time series plot function to get a figure and then update the plot?

jeffreypaul15 replied:
Why though? I'm a bit confused

nabobalis replied:
It might some code that could be useful?

jeffreypaul15 replied:
So what would the generic time series plot function produce? As of now it just plots the timeseries dataframe and isn't specific to any source right

nabobalis replied:
Well if the generic one creates the plot for us, we can call it and then update the figure?

jeffreypaul15 replied:
Hmm I suppose we could do that as well. Would that be better? I'll re-structure this PR then.

jeffreypaul15 replied:
I don't suppose, `eve esp` and `lyra` would be exceptions because as I mentioned above, they contain multiple axes so we can't call plot which deals with just one axes. Correct me if I'm wrong, I haven't gone into too much depth.

nabobalis replied:
Not sure.

jeffreypaul15 replied:
Nope, I was wrong, setting `subplot=True` would fix the multiple axes problem. Shall I restructure this then?

nabobalis replied:
Sure.

jeffreypaul15 commented:
@nabobalis On continuing with using the source having to call the main class to plot the figure, we get 
`UserWarning: To output multiple subplots, the figure containing the passed axes is being cleared`
The reason is because generic time series just fetches the axes for us and then returns the axes to the source where I'm plotting it. 
Do we just ignore this warning?

nabobalis commented:
I would like to avoid rasing it if possible

jeffreypaul15 commented:
> I would like to avoid rasing it if possible

As of now it seems to be unavoidable, another problem that I've stumbled upon is that the `noaa`source creates it's axes from it's dataframe like any other source but only certain parts of the entire dataframe is plotted (columns based on user input). Calling plot() here would be redundant as we'd have extra `2D lines` which aren't what is required by the user.
Based on this and the previous warning issue with multiple plots I suggest that it'd be better to not call generic time series to plot, and instead leave that functionality to a source-less timeseries object.

nabobalis commented:
Sure.

jeffreypaul15 commented:
> Sure.

Should the docstrings have examples?

nabobalis commented:
No its ok.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,11 +57,14 @@ class ESPTimeSeries(GenericTimeSeries):
-    def peek(self, title='EVE/ESP Level1', **kwargs):
+    def plot(self, axes=None, title="EVE/ESP Level1", **kwargs):
         """
+        Plots the EVE ESP Leve1 timeseries data.
```
Does plot not return the figure/axes?

jeffreypaul15 replied:
No, do we want them to return the axes?

ayshih replied:
Yes

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,11 +57,14 @@ class ESPTimeSeries(GenericTimeSeries):
         Parameters
         ----------
+        axes : `matplotlib.axes.Axes`
+            The axes on which to plot the TimeSeries. Defaults to current axes.
```
Now we need a return section in the doc string.

jeffreypaul15 replied:
Ah, will do that.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,63 @@ class ESPTimeSeries(GenericTimeSeries):
+
+        Returns
+        -------
+        axes : `~matplotlib.axes.Axes`
```
```suggestion
        `~matplotlib.axes.Axes`
```

jeffreypaul15 replied:
Is this the format that's followed everywhere?
Won't we need to change it here as well? https://github.com/sunpy/sunpy/blob/3940d01382c1863e3250b5cc7f96531265b29c43/sunpy/timeseries/timeseriesbase.py#L462

nabobalis replied:
It is the format that should be used.

jeffreypaul15 replied:
Alright then, I'll make that change.

jeffreypaul15 commented:
@nabobalis Tests seem to be unrelated issues?

nabobalis commented:
Yeah.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,63 @@ class ESPTimeSeries(GenericTimeSeries):
-    def peek(self, title='EVE/ESP Level1', **kwargs):
+    def plot(self, axes=None, title="EVE/ESP Level1", **kwargs):
         """
+        Plots the EVE ESP Leve1 timeseries data.
```
```suggestion
        Plots the EVE ESP Level 1 timeseries data.
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,63 @@ class ESPTimeSeries(GenericTimeSeries):
 
-    @peek_show
-    def peek(self, title='EVE/ESP Level1', **kwargs):
+    def plot(self, axes=None, title="EVE/ESP Level1", **kwargs):
```
```suggestion
    def plot(self, axes=None, title="EVE/ESP Level 1", **kwargs):
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,63 @@ class ESPTimeSeries(GenericTimeSeries):
+
         Parameters
         ----------
+        axes : `matplotlib.axes.Axes`
```
```suggestion
        axes : `matplotlib.axes.Axes`, optional
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,63 @@ class ESPTimeSeries(GenericTimeSeries):
         Parameters
         ----------
+        axes : `matplotlib.axes.Axes`
+            The axes on which to plot the TimeSeries. Defaults to current axes.
```
Is "Defaults to current axes." true?

jeffreypaul15 replied:
Sorry, this should be changed.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,63 @@ class ESPTimeSeries(GenericTimeSeries):
         self._validate_data_for_plotting()
 
-        names = ('Flux \n 0.1-7nm', 'Flux \n 18nm', 'Flux \n 26nm', 'Flux \n 30nm', 'Flux \n 36nm')
+        if axes is None:
```
So this function will do nothing if you provide an axes?

nabobalis replied:
We might need some plot tests after all.

jeffreypaul15 replied:
We'd need to provide it with a list of axes, should I mention that in the docstring?

nabobalis replied:
Maybe.

jeffreypaul15 replied:
As a note? Or should under parameters would do?

nabobalis replied:
Would a list of axes change anything?

jeffreypaul15 replied:
This source in particular requires a list of axes right so not passing a list would return an index out of range error, I suppose I'd have to handle that error in the code?

nabobalis replied:
Sure. I think we need plot tests for each source to ensure it works as we want. They won't need to be figure tests. We can leave that to peek.

jeffreypaul15 replied:
How would we test these plots without a figure test?

nabobalis replied:
We check the inputs work as we expect them to?

jeffreypaul15 replied:
I'm a bit confused as to how exactly we'd be testing the axes input. 
How would we vary the inputs to test?

nabobalis replied:
Its fine, lets not test this.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,63 @@ class ESPTimeSeries(GenericTimeSeries):
+        **kwargs : `dict`
+            Additional plot keyword arguments that are handed to `pandas.DataFrame.plot`.
+        """
+        self._validate_data_for_plotting()
+        axes = self.to_dataframe().plot(subplots=True, sharex=True, **kwargs)
```
This are in the plot function, so I don't see why we can't remove them here?

jeffreypaul15 replied:
Sorry, must've forgotten

jeffreypaul15 commented:
@nabobalis The eve, lyra, noaa sources all plot from a pandas dataframe exclusively and each source has their own number of columns. For these particular sources, we can't pass axes as an argument so I have removed that. We don't need tests for these sources as they don't have axes as arguments. 
The remaining sources (rhessi, norh, goes and fermi_gbm) work similar to how GenericMap works and `plt.gca()` can be used, thereby allowing for axes to be passed via :
```python
fig, ax = plt.subplots()
axs = ts.plot(axes=ax)
```
I have written tests to confirm if the returned axes are `matplotlib.axes._subplots.AxesSubplot` but I don't know if this is required. If not I can get rid of them.

nabobalis commented:
> I have written tests to confirm if the returned axes are `matplotlib.axes._subplots.AxesSubplot` but I don't know if this is required. If not I can get rid of them.

It would be useful to check that the input axes gets updated by plot.

jeffreypaul15 commented:
> It would be useful to check that the input axes gets updated by plot.

Is there a simple way to check for that? This is where I'm a bit confused.

nabobalis commented:
Not sure.

jeffreypaul15 commented:
> Not sure.

How about checking for the title change in `ax.title`, if the input axes is updated the title is also updated?

nabobalis commented:
If it works.

jeffreypaul15 commented:
> If it works.

Should be fine now. I've added `title` arguments to the sources that didn't have them as well.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
         ----------
         title : `str`, optional
-            Plot title.
+            The title of the plot. Defaults to "EVE/ESP Level1".
```
```suggestion
            The title of the plot. Defaults to "EVE/ESP Level 1".
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
         title : `str`, optional
-            Plot title.
+            The title of the plot. Defaults to "EVE/ESP Level 1".
         **kwargs : `dict`
```
I always forget if this need an optional.

jeffreypaul15 replied:
Other parts of the code follow the same behavior so I guess we do it here too?

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
+        Parameters
+        ----------
+        title : `str`, optional
+            The title of the plot. Defaults to "EVE/ESP Level1".
```
```suggestion
            The title of the plot. Defaults to "EVE/ESP Level 1".
```

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,55 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
+        axes.set_yscale("log")
+        if title is None:
+            axes.set_title('Fermi GBM Summary data ' + str(self.meta.get(
+                'DETNAM').values()))
```
This is a strange line break?

jeffreypaul15 replied:
Not sure why this happened. I'll fix this

jeffreypaul15 commented:
Test failures look like unrelated issues.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
+        `~matplotlib.axes.Axes`
+            The plot axes.
         """
 
```
Can you remove all these blank lines?

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
 
         self._validate_data_for_plotting()
 
-        names = ('Flux \n 0.1-7nm', 'Flux \n 18nm', 'Flux \n 26nm', 'Flux \n 30nm', 'Flux \n 36nm')
```
Did this have to move?

jeffreypaul15 replied:
I moved it a line below, I don't suppose that changes anything right?

nabobalis replied:
No but why?

jeffreypaul15 replied:
Looked neater that way, no worries, I'll move it back.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
+    @peek_show
+    def peek(self, title="EVE/ESP Level 1", **kwargs):
+        """
+        Plots the EVE ESP Level 1 timeseries data by calling
+        `~sunpy.timeseries.sources.eve.ESPTimeSeries.plot`.
```
```suggestion
        Displays the EVE ESP Level 1 timeseries data by calling
        `~sunpy.timeseries.sources.eve.ESPTimeSeries.plot`.
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
         fig = axes[0].get_figure()
         fig.tight_layout()
         fig.subplots_adjust(hspace=0.05)
+
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,59 @@ class ESPTimeSeries(GenericTimeSeries):
+        **kwargs : `dict`
+            Additional plot keyword arguments that are handed to `pandas.DataFrame.plot`.
+        """
+
```
```suggestion
```

jeffreypaul15 replied:
I've removed all extra lines locally, I shall commit separately.

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,53 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
 
+    def plot(self, axes=None, title=None, **kwargs):
+        """
+        Plots the GBM lightcurve TimeSeries with data from its pandas dataframe.
```
```suggestion
        Plots the GBM lightcurve timeseries with data from its pandas dataframe.
```

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,53 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
-        Plots the GBM lightcurve TimeSeries. An example can be seen below:
+        Plots the GBM lightcurve TimeSeries by calling
+        `~sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries.plot`.
+        An example can be seen below:
```
```suggestion
```

jeffreypaul15 replied:
Every `peek()` method has this, do we want to get rid of this line?

nabobalis replied:
Sure.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,35 +57,53 @@ class ESPTimeSeries(GenericTimeSeries):
         fig = axes[0].get_figure()
         fig.tight_layout()
         fig.subplots_adjust(hspace=0.05)
+
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -175,7 +193,7 @@ class EVESpWxTimeSeries(GenericTimeSeries):
-        Plots the time series in a new figure. An example is shown below:
+        Plots the time series in a new figure.
 
         ..
```
Can you remove this and see if it works now?

jeffreypaul15 replied:
The plot? It works right

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,50 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
+        `~matplotlib.axes.Axes`
+            The plot axes.
+        """
+
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,50 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
+        """
+
+        self._validate_data_for_plotting()
+
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,50 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
+        axes.set_xlabel('Start time: ' + self.to_dataframe().index[0].strftime('%Y-%m-%d %H:%M:%S UT'))
+        axes.set_ylabel('Counts/s/keV')
+        axes.legend()
+
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,50 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
-        Plots the GBM lightcurve TimeSeries. An example can be seen below:
+        Displays the GBM lightcurve timeseries by calling
+        `~sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries.plot`.
+        An example can be seen below:
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -77,22 +117,8 @@ def peek(self, **kwargs):
-        ax.legend()
+        self.plot(axes=ax, **kwargs)
         fig.autofmt_xdate()
 
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/goes.py:
```diff
@@ -64,63 +64,87 @@ class XRSTimeSeries(GenericTimeSeries):
-        ax.xaxis.set_major_formatter(formatter)
+        formatter = matplotlib.dates.DateFormatter("%H:%M")
+        axes.xaxis.set_major_formatter(formatter)
+
```
```suggestion

```

nabobalis commented on sunpy/timeseries/sources/goes.py:
```diff
@@ -64,63 +64,87 @@ class XRSTimeSeries(GenericTimeSeries):
+        axes.xaxis.grid(False, "major")
+        axes.legend()
 
         # TODO: display better tick labels for date range (e.g. 06/01 - 06/05)
```
Can we do this?

jeffreypaul15 replied:
The resultant plots are the same so I suppose we can right

nabobalis commented on sunpy/timeseries/sources/goes.py:
```diff
@@ -64,63 +64,87 @@ class XRSTimeSeries(GenericTimeSeries):
         ax2.yaxis.set_minor_locator(mticker.FixedLocator(centers))
         ax2.set_yticklabels(labels, minor=True)
         ax2.set_yticklabels([])
 
```
```suggestion
```

nabobalis commented on sunpy/timeseries/sources/goes.py:
```diff
@@ -64,63 +64,87 @@ class XRSTimeSeries(GenericTimeSeries):
+        fig, ax = plt.subplots()
+        self.plot(axes=ax, title=title, **kwargs)
         fig.autofmt_xdate()
 
```
```suggestion

```

jeffreypaul15 replied:
Do we not need an empty line between the method and it's return statement?

nabobalis replied:
No?

jeffreypaul15 replied:
Oh wow okay, I had been doing that for every PR that I had made until now. I'll make this change.

nabobalis replied:
It's just a style choice.

nabobalis commented on sunpy/timeseries/sources/lyra.py:
```diff
@@ -54,46 +54,68 @@ class LYRATimeSeries(GenericTimeSeries):
         """
-        Plots the LYRA data. An example is shown below:
-
-        .. plot::
-
-            import sunpy.timeseries
-            import sunpy.data.sample
-            lyra = sunpy.timeseries.TimeSeries(sunpy.data.sample.LYRA_LEVEL3_TIMESERIES, source='LYRA')
-            lyra.peek()
```
Restore this?

jeffreypaul15 replied:
It's part of peek in the method below, this is plot()

jeffreypaul15 commented:
@nabobalis Do I need to rebase this as well to check for figure tests?

nabobalis commented:
Sure.

nabobalis commented:
Is this ready now you think?

jeffreypaul15 commented:
> Is this ready now you think?

I seem to have messed up the git commands here while rebasing. My aplogies.

jeffreypaul15 commented:
@nabobalis The new figure tests revealed that the previous peek method had title = 
![image](https://user-images.githubusercontent.com/50923538/117155018-0bf4cb00-adda-11eb-81c1-e30987bec96d.png)
and the updated one has : 
![image](https://user-images.githubusercontent.com/50923538/117155070-1747f680-adda-11eb-8309-9c38dce1ff33.png)

That's the reason for change in the figure hashes.

nabobalis commented:
Sure.

jeffreypaul15 commented:
> Sure.

The failing test seems to be unrelated, everything else seems good to go I guess.

samaloney commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,46 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
 
+    def plot(self, axes=None, title=None, **kwargs):
+        """
+        Plots the GBM lightcurve timeseries with data from its pandas dataframe.
```
```suggestion
        Plots the GBM timeseries with data from its pandas dataframe.
```

samaloney commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,46 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
     def peek(self, **kwargs):
         """
-        Plots the GBM lightcurve TimeSeries. An example can be seen below:
+        Displays the GBM lightcurve timeseries by calling
```
```suggestion
        Displays the GBM timeseries by calling
```

samaloney commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -175,15 +191,14 @@ class EVESpWxTimeSeries(GenericTimeSeries):
     @peek_show
     def peek(self, column=None, **kwargs):
         """
-        Plots the time series in a new figure. An example is shown below:
+        Plots the time series in a new figure.
 
-        ..
-            .. plot::
+        .. plot::
 
-                import sunpy.timeseries
-                from sunpy.data.sample import EVE_TIMESERIES
-                eve = sunpy.timeseries.TimeSeries(EVE_TIMESERIES, source='eve')
-                eve.peek(subplots=True)
+            import sunpy.timeseries
+            from sunpy.data.sample import EVE_TIMESERIES
+            eve = sunpy.timeseries.TimeSeries(EVE_TIMESERIES, source='eve')
+            eve.peek(subplots=True)
```
For some reason this doesn't seem to be showing up in the built version of the [docs]( https://sunpy--5200.org.readthedocs.build/en/5200/api/sunpy.timeseries.sources.ESPTimeSeries.html#sunpy.timeseries.sources.ESPTimeSeries) like the rest

jeffreypaul15 replied:
What do you mean?

dstansby replied:
I think we should just get rid of these plot example lines, we don't have them for other `peek()` methods.

jeffreypaul15 replied:
All of them have examples, esp is the only one without

nabobalis replied:
Can you check that each example shows up and looks nice?

jeffreypaul15 replied:
https://sunpy--5200.org.readthedocs.build/en/5200/api/sunpy.timeseries.sources.EVESpWxTimeSeries.html#sunpy.timeseries.sources.EVESpWxTimeSeries
Looks bad, the others are fine I suppose

nabobalis replied:
I think we should address this.

jeffreypaul15 replied:
The figure has too much info to fit in standard figsize, either we increase the figsize which would be inconsistent in the docs, or plot the figures with a few columns (column argument already exists), I'd say the latter approach is better

nabobalis replied:
Why not make it bigger? 

Can we also not change the subplots to use the same x axis?

jeffreypaul15 replied:
I thought making it bigger would make it inconsistent with the other plots? If not then we could do that.
> Can we also not change the subplots to use the same x axis?

Yeah, `sharex=True` argument isn't passed this would case them to use the same x axis.

nabobalis replied:
They don't have to be the same size, they display different amounts of data.

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,32 +57,48 @@ class ESPTimeSeries(GenericTimeSeries):
+        return axes
 
+    @peek_show
+    def peek(self, title="EVE/ESP Level 1", **kwargs):
```
Does peek need to take title?

jeffreypaul15 replied:
I don't see an issue here right? Do we not want that?

nabobalis replied:
Why would we want it on peek?

nabobalis replied:
I do wonder if we want it on plot as well.

nabobalis replied:
I think at the very least, peek shouldn't take any arguments.

dstansby replied:
Lets leave this for now (we shouldn't change the signature without deprecating anyway), we can always revisit the signature of `peek()` in another issue or PR.

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,46 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
     # Class attribute used to specify the source class of the TimeSeries.
     _source = 'gbmsummary'
 
+    def plot(self, axes=None, title=None, **kwargs):
```
I feel like the API on plot should be the same for all?

jeffreypaul15 replied:
This changes depending on the number of axes right? This source has only one and isn't deriving it from it's pandas dataframe while EVE/ESP and a few others are

nabobalis replied:
Sure but I guess on some level I am uncomfortable for a function that is named the same on a subclass of time series to a different API.

I don't really have a good answer to this.

jeffreypaul15 replied:
They still return the axes similar to the others right so I thought it wouldn't matter but if that's the case then I'm open to removing the plot functions where the axes are not singular

nabobalis replied:
Removing?

jeffreypaul15 replied:
I meant keeping removing it as it was before this PR

nabobalis replied:
Sorry, I've lost track. Removing the axes keywords in plot?

jeffreypaul15 replied:
Removing this method altogether, if that might be better (for those with multiple axes, but 3 of them will have multiple axes if we decide NOAA to be one as well)

nabobalis replied:
You mean removing plot? I think we want to keep plot, in future I plan to push to remove peek.

jeffreypaul15 replied:
Ah okay, then I shall wait for a better solution to this

dstansby replied:
I think we should remove the title argument, but leave axes.

nabobalis commented on sunpy/timeseries/sources/noaa.py:
```diff
@@ -59,60 +60,89 @@ class NOAAIndicesTimeSeries(GenericTimeSeries):
 
-    @peek_show
-    def peek(self, type='sunspot SWO', **kwargs):
+    def plot(self, title='Solar Cycle Progression', plot_type='sunspot SWO', dataframe=None, **kwargs):
```
I know I wrote this but I do wonder if this should just output all of the plot types in big subplot.

jeffreypaul15 replied:
Should I make that change?

nabobalis replied:
It might be, I will let others have a review and decide.

jeffreypaul15 commented on sunpy/timeseries/sources/rhessi.py:
```diff
@@ -136,59 +136,77 @@ class RHESSISummaryTimeSeries(GenericTimeSeries):
         self._validate_data_for_plotting()
-
+        if axes is None:
+            axes = plt.gca()
```
Sure, I'll add that in

jeffreypaul15 replied:
@nabobalis `return fig` from every timeseries `peek()` method is tested via their figure tests right?
I haven't altered those figure test, yet codecov says that the coverage for `return fig` from every source is removed?
That's odd

nabobalis replied:
I think that is correct since the return is captured by the decorator?

jeffreypaul15 replied:
I'm sorry I don't follow, so how has the coverage been altered when I'm still returning `fig`?

nabobalis replied:
Don't know.

nabobalis commented on sunpy/timeseries/timeseriesbase.py:
```diff
@@ -461,7 +461,7 @@ def plot(self, axes=None, **plot_args):
 
```
I can't highlight the line, but it says above the `**plot_args` are passed to :meth:`pandas.DataFrame.plot`. But is that true?

jeffreypaul15 replied:
`self._data` returns a `pandas.core.frame.DataFrame`, so arguments passed to `self._data.plot()` are sent to pandas

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,32 +57,48 @@ class ESPTimeSeries(GenericTimeSeries):
 
-    @peek_show
-    def peek(self, title='EVE/ESP Level1', **kwargs):
+    def plot(self, title="EVE/ESP Level 1", **kwargs):
```
On the axes in plot, the original plot method on time series base takes an axes and I think each plot method should as a result at least keep that as a named argument. The idea is to keep the API uniform, we can add title after the fact but I am also leaning toward removing title as an arg fully.

Thoughts?

jeffreypaul15 replied:
I guess that makes sense but the user should pass the axes from a pandas dataframe, else this won't really behave the same way it does when `ax` from `fig, ax = plt.subplots()` is passed to it directly, and `plt.gca` won't work either

dstansby replied:
The issue is that the plots created have multiple axes, so it's not possible to have a single `axes` argument here. I would remove the title argument though, and just always set the title to be what is currently the default. It's easy enough for the user to override this themselves after calling `.plot()`.

dstansby commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,32 +57,48 @@ class ESPTimeSeries(GenericTimeSeries):
 
+        Returns
+        -------
+        `~matplotlib.axes.Axes`
```
This actually returns an array of `Axes`.
```suggestion
        array of `~matplotlib.axes.Axes`
```

dstansby commented:
I think it's worth holding off on this until we've discussed what signature new `plot()` functions should have at the weekly meeting this week. Once we've discussed I'll post another comment about what we decided.

nabobalis commented:
Discussion in community meeting:

Make sure we don't add or break/remove the original plot/peek api for timeseries. 
Move all custom plotting to plot as required. 

We would need to deal with plots that have multiple axes but I think we can handle that.

jeffreypaul15 commented:
> We would need to deal with plots that have multiple axes but I think we can handle that.

nabobalis commented:
> > Discussion in community meeting:
> > Make sure we don't add or break/remove the original plot/peek api for timeseries.
> > Move all custom plotting to plot as required.
> > We would need to deal with plots that have multiple axes but I think we can handle that.
> 
> So, just removing the title arguments?

If thats all the difference is, then yes.

jeffreypaul15 commented:
> If thats all the difference is, then yes.

Do we still have axes as an argument? Because of the multiple axes issue

nabobalis commented:
> > If thats all the difference is, then yes.
> 
> Do we still have axes as an argument? Because of the multiple axes issue

If its part of the original API, yes.

jeffreypaul15 commented:
> > > If thats all the difference is, then yes.
> > 
> > 
> > Do we still have axes as an argument? Because of the multiple axes issue
> 
> If its part of the original API, yes.

Then I'll have to add the axes back to the sources where there are multiple axes which are being derived from a pandas dataframe, otherwise I don't think there are any changes other than the original plot method now producing the required plot instead of the one in timeseries base

jeffreypaul15 commented:
When we have multiple axes, how do we handle the situation then? For now any axes passed will be overwritten by 
`axes = self.to_dataframe().plot(subplots=True, sharex=True, **kwargs)`
Which is wrong, but I'm not really sure how to proceed with this.

jeffreypaul15 commented:
Do we keep the title argument in peek? Since peek is calling plot do we get rid of the title here as well?

nabobalis commented:
> Do we keep the title argument in peek? Since peek is calling plot do we get rid of the title here as well?

if title was a keyword taken in peek, it stays.

jeffreypaul15 commented:
@nabobalis The multiple axes issue shouldn't be a problem now, although if the user does:
```py
eve = ts.TimeSeries(esp_filepath)
fig, ax = plt.subplots(5,1)
eve.plot(axes=ax)
```
![image](https://user-images.githubusercontent.com/50923538/119085303-22796400-ba21-11eb-8060-88f0d78a0177.png)

the colours aren't in accordance with when the user doesn't pass any axes, we could fix this by having a colours array but that is specific to this dataframe and I don't suppose that's right to do

```py
eve.plot()
```
produces :
![image](https://user-images.githubusercontent.com/50923538/119085479-6ff5d100-ba21-11eb-91e9-206b85c93569.png)

nabobalis commented:
> the colours aren't in accordance with when the user doesn't pass any axes as we're plotting each column of the dataframe separately on each of the axes passed, we could fix this by having a colours array but that is specific to this dataframe and I don't suppose that's right to do

Why?

nabobalis commented on sunpy/timeseries/sources/eve.py:
```diff
@@ -57,32 +57,55 @@ class ESPTimeSeries(GenericTimeSeries):
+        """
         self._validate_data_for_plotting()
+        names = ("Flux \n 0.1-7nm", "Flux \n 18nm", "Flux \n 26nm", "Flux \n 30nm", "Flux \n 36nm")
+        if isinstance(axes, np.ndarray):
+            data_ = self.to_dataframe()
+            for i, ax in enumerate(axes):
+                ax.plot(data_[data_.columns[i]], label=data_.columns[i])
+                ax.set_ylabel(names[i])
+                ax.legend(loc="upper right")
+        elif axes is None:
+            axes = self.to_dataframe().plot(subplots=True, sharex=True, **kwargs)
+            axes[-1].set_xlim(self.to_dataframe().index[0], self.to_dataframe().index[-1])
+            for i, ax in enumerate(axes):
+                ax.set_ylabel(names[i])
+                ax.legend(loc="upper right")
```
Can this block not be made simpler?

jeffreypaul15 commented:
> 
> 
> > the colours aren't in accordance with when the user doesn't pass any axes as we're plotting each column of the dataframe separately on each of the axes passed, we could fix this by having a colours array but that is specific to this dataframe and I don't suppose that's right to do
> 
> Why?

I was under the assumption that this would be used for a variety of dataframes where the columns might exceed the default number of colours that we specify right?

nabobalis commented:
> > > the colours aren't in accordance with when the user doesn't pass any axes as we're plotting each column of the dataframe separately on each of the axes passed, we could fix this by having a colours array but that is specific to this dataframe and I don't suppose that's right to do
> > 
> > 
> > Why?
> 
> I was under the assumption that this would be used for a variety of dataframes where the columns might exceed the default number of colours that we specify right?

Maybe?

jeffreypaul15 commented:
> 
> 
> > > > the colours aren't in accordance with when the user doesn't pass any axes as we're plotting each column of the dataframe separately on each of the axes passed, we could fix this by having a colours array but that is specific to this dataframe and I don't suppose that's right to do
> > > 
> > > 
> > > Why?
> > 
> > 
> > I was under the assumption that this would be used for a variety of dataframes where the columns might exceed the default number of colours that we specify right?
> 
> Maybe?

`dataframe.plot()` handles this by itself, including the rotation of the `xticklabels` so by plotting it separately we would need to manually set all of this

nabobalis commented:
> > > > > the colours aren't in accordance with when the user doesn't pass any axes as we're plotting each column of the dataframe separately on each of the axes passed, we could fix this by having a colours array but that is specific to this dataframe and I don't suppose that's right to do
> > > > 
> > > > 
> > > > Why?
> > > 
> > > 
> > > I was under the assumption that this would be used for a variety of dataframes where the columns might exceed the default number of colours that we specify right?
> > 
> > 
> > Maybe?
> 
> `dataframe.plot()` handles this by itself, including the rotation of the `xticklabels` so by plotting it separately we would need to manually set all of this

I guess we add it then?

jeffreypaul15 commented:
> I guess we add it then?

I've added the exact colours that are cycled and also updated the docs figure to be of a `figsize=(20,10)`, the plot should now be rendered properly.

jeffreypaul15 commented:
@nabobalis the failing tests seem unrelated right?

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -59,10 +59,40 @@ class GBMSummaryTimeSeries(GenericTimeSeries):
+        return axes
+
     @peek_show
     def peek(self, **kwargs):
```
This is lacking a title argument, if all the other peeks have one.

jeffreypaul15 replied:
Didn't we decide to not add one?
Do I add it in xD

nabobalis replied:
I don't see why some peek methods should have it and some don't.

jeffreypaul15 replied:
Yeah, the some of the peek methods had it initially as well, I thought we wouldn't be changing how it was before, so I left it in

nabobalis replied:
I think we wanted to not break the old API but we should also try to unify them where possible.

jeffreypaul15 replied:
> > > Discussion in community meeting:
> > > Make sure we don't add or break/remove the original plot/peek api for timeseries.
> > > Move all custom plotting to plot as required.
> > > We would need to deal with plots that have multiple axes but I think we can handle that.
> > 
> > 
> > So, just removing the title arguments?
> 
> If thats all the difference is, then yes.

Okay, so add title arguments for every peek method but not plot right?

nabobalis replied:
Pretty much.

jeffreypaul15 replied:
Alright, I'll add that in.

nabobalis replied:
Thanks.

nabobalis commented on sunpy/timeseries/sources/noaa.py:
```diff
@@ -205,11 +227,42 @@ class NOAAPredictIndicesTimeSeries(GenericTimeSeries):
+        return axes
+
     @peek_show
     def peek(self, **plot_args):
```
Same here with the title arg.

nabobalis commented on sunpy/timeseries/sources/norh.py:
```diff
@@ -54,10 +54,42 @@ class NoRHTimeSeries(GenericTimeSeries):
+        return axes
+
     @peek_show
     def peek(self, **kwargs):
```
This one as well. Did we not set this on enough peek methods?

jeffreypaul15 replied:
I think we decided that we wouldn't be adding it in if it had it before I think

jeffreypaul15 commented:
@nabobalis A few sources (lyra and fermi_gbm) derive their title from the dataframe :
`axes[0].set_title("LYRA ({0:{1}})".format(self.to_dataframe().index[0], TIME_FORMAT))`
Do we still accept a default `title` argument here?

nabobalis commented:
> @nabobalis A few sources (lyra and fermi_gbm) derive their title from the dataframe :
> `axes[0].set_title("LYRA ({0:{1}})".format(self.to_dataframe().index[0], TIME_FORMAT))`
> Do we still accept a default `title` argument here?

Sure.

nabobalis commented on sunpy/timeseries/sources/fermi_gbm.py:
```diff
@@ -73,28 +103,18 @@ def peek(self, **kwargs):
-        # Check we have a timeseries valid for plotting
-        self._validate_data_for_plotting()
-
+        if title:
```
```suggestion
        if title is None:
```

nabobalis commented on sunpy/timeseries/sources/lyra.py:
```diff
@@ -54,49 +56,75 @@ class LYRATimeSeries(GenericTimeSeries):
+            :meth:`pandas.DataFrame.plot`.
+        """
+        axes = self.plot(names=names, **kwargs)
+        if title:
```
```suggestion
        if title is None:
```

nabobalis commented on sunpy/timeseries/sources/noaa.py:
```diff
@@ -56,60 +57,83 @@ class NOAAIndicesTimeSeries(GenericTimeSeries):
 
-    @peek_show
-    def peek(self, type='sunspot SWO', **kwargs):
+    def plot(self, axes=None, plot_type='sunspot SWO', **kwargs):
```
The change to "plot_type" will need a breaking changelog entry.

jeffreypaul15 replied:
I changed this because type is a keyword, should I revert to type?

nabobalis replied:
No, it shouldn't have been `type` in the first place.

nabobalis commented:
Can this be squash merged and rebased please?

nabobalis commented on changelog/5200.breaking.rst:
```diff
@@ -0,0 +1 @@
+:meth:`sunpy.timeseries.sources.NOAAIndicesTimeSeries.peek` accepts ``plot_type`` as an argument instead of ``type``.
```
```suggestion
:meth:`sunpy.timeseries.sources.NOAAIndicesTimeSeries.plot` accepts ``plot_type`` as an argument instead of ``type``.
```

jeffreypaul15 replied:
peek as well right?

nabobalis replied:
Did `peek` take the argument before?

jeffreypaul15 replied:
yeah it did

nabobalis replied:
My bad.

nabobalis commented:
I wanted to check what plot vs peek looked light right now.
Each plot is plot then peek.

# noaa_json_pre
![noaa_json_pre_plot](https://user-images.githubusercontent.com/1392107/122890270-71c1f580-d33b-11eb-9e1b-c047a41b6775.png)
![noaa_json_pre_peek](https://user-images.githubusercontent.com/1392107/122890307-75557c80-d33b-11eb-9659-f1351161501a.png)

# noaa_pre_json
![noaa_pre_json_peek](https://user-images.githubusercontent.com/1392107/122890274-725a8c00-d33b-11eb-9aa2-7e4e008bb872.png)
![noaa_pre_json_plot](https://user-images.githubusercontent.com/1392107/122890277-725a8c00-d33b-11eb-918d-93274b8e67e8.png)

# norh
norh plot broke for me but peek worked.
![norh_peek](https://user-images.githubusercontent.com/1392107/122890281-72f32280-d33b-11eb-9aa4-d763fe1098e1.png)

# rhessi
![rhessi_plot](https://user-images.githubusercontent.com/1392107/122890287-738bb900-d33b-11eb-9a05-dd98c42395f2.png)
![rhessi_peek](https://user-images.githubusercontent.com/1392107/122890285-72f32280-d33b-11eb-99a9-7075ae00390a.png)

# esp
![esp_plot](https://user-images.githubusercontent.com/1392107/122890293-738bb900-d33b-11eb-8625-fbcaedc496c1.png)
![esp_peek](https://user-images.githubusercontent.com/1392107/122890290-738bb900-d33b-11eb-8907-0cd1e8496182.png)

# eve
![eve_plot](https://user-images.githubusercontent.com/1392107/122890296-74244f80-d33b-11eb-8492-a2cebec59957.png)
![eve_peek](https://user-images.githubusercontent.com/1392107/122890294-74244f80-d33b-11eb-8d0a-7b441f6b3e74.png)

# fermi_gbm
![fermi_gbm_plot](https://user-images.githubusercontent.com/1392107/122890299-74bce600-d33b-11eb-9096-2b784a4a57e9.png)
![fermi_gbm_peek](https://user-images.githubusercontent.com/1392107/122890298-74244f80-d33b-11eb-80b0-1d8da32f4365.png)

# goes
![goes_plot](https://user-images.githubusercontent.com/1392107/122890302-74bce600-d33b-11eb-9a5a-9931742053d3.png)
![goes_peek](https://user-images.githubusercontent.com/1392107/122890300-74bce600-d33b-11eb-8d56-6280e0290e3c.png)

# lyra
![lyra_plot](https://user-images.githubusercontent.com/1392107/122890306-75557c80-d33b-11eb-9522-021139bb31d5.png)
![lyra_peek](https://user-images.githubusercontent.com/1392107/122890304-75557c80-d33b-11eb-90cc-7851dfcda81e.png)

There are still some big differences? Is there anyway to resolve this?

nabobalis reviewed (changes requested):
Undoing my approval.

jeffreypaul15 commented:
@nabobalis These differences are resolved in this PR right?
The peek and plot all look alike for me

nabobalis commented:
I will double check I had checked out the correct code but this was from testing this PR.

jeffreypaul15 commented:
> I will double check I had checked out the correct code but this was from my attempt at testing this PR.

Sure, the current release version has the same differences that you've mentioned

nabobalis commented:
I had sunpy installed twice in my env and this created the conflict.

nabobalis commented:
Thanks again @jeffreypaul15
